.gitlab-ci-common.yml

```diff
@@ -0,0 +1,52 @@
+# .gitlab-ci-common.yml
+#
+# This gitlab-ci file defines shell helper functions to be used by any Mender
+# pipeline. After including the file, you need to source the different snippets.
+#
+# Add it to the project in hand through Gitlab's include functionality:
+#
+# include:
+#   - project: 'Northern.tech/Mender/mendertesting'
+#     file: '.gitlab-ci-common.yml'
+#
+# And then source the desired functions in your job:
+#
+# my-fantastic-job:
+#   before_script:
+#     - *mender_ci_save_tmp_artifact
+#
+# Read the documentation for each function for more details
+#
+
+
+# Usage: mender_ci_save_tmp_artifact file-to-save
+#
+# Saves the given file in S3 bucket temporary storage.
+# For security, it computes the SHA256 checksum and saves it in
+# ${CI_PROJECT_DIR}/checksums. It is responsibility of the caller to save the
+# directory as part of the CI job artifacts:
+#   artifacts:
+#     paths:
+#       - checksums
+# Later jobs that will call mender_ci_load_tmp_artifact have to define "needs:"
+# or "depends:" to load the checksums directory
+.mender_ci_save_tmp_artifact: &mender_ci_save_tmp_artifact |
+  function mender_ci_save_tmp_artifact {
+    if [ -z "$1" ] then; echo Usage $0 file-to-save; exit 1; fi
+    user_file="$1"
+    mkdir -p ${CI_PROJECT_DIR}/checksums
+    sha256sum ${user_file} > ${CI_PROJECT_DIR}/checksums/$(basename ${user_file}).sha256
+    aws s3 cp ${user_file} s3://mender-gitlab-tmp-storage/$CI_PROJECT_NAME/$CI_PIPELINE_ID/$(basename ${user_file})
+  }
+
+# Usage: mender_ci_load_tmp_artifact file-to-load
+#
+# Loads the given file from S3 bucket temporary storage. It checks the SHA256
+# checksum (see mender_ci_save_tmp_artifact).
+.mender_ci_load_tmp_artifact: &mender_ci_load_tmp_artifact |
+  function mender_ci_load_tmp_artifact {
+    if [ -z "$1" ] then; echo Usage $0 file-to-load; exit 1; fi
+    user_file="$1"
+    aws s3 cp s3://mender-gitlab-tmp-storage/$CI_PROJECT_NAME/$CI_PIPELINE_ID/$(basename ${user_file}) ${user_file}
+    sha256sum -c ${CI_PROJECT_DIR}/checksums/$(basename ${user_file}).sha256
```

Comment:
that might be useful to also get from s3 in order to compare it here

Reply:
For security reasons, we need to have it in the GitLab pipeline. What this is trying to protect is against someone running a different pipeline and tampering your files, as the keys for this temporary bucket are not protected. See https://github.com/mendersoftware/mender-convert/commit/9d18a8717d8812dba135c3e2231e786afeb1ac6c

Reply:
oh ok, so I guess then the checksum comparison is left to the outside pipeline, I thought this was to be contained in these functions... (besides also expecting this to be used across pipelines)

Reply:
It is tricky, but is the only way we think it can be safe. The checksum comparison is done by these functions (note `sha256sum -c`), the only left to the outside pipeline is to collect the `checksums/` as job artifacts.